ludwig/features/timeseries_feature.py

```diff
@@ -103,21 +120,93 @@ def pad(vector):
             limit = min(vector.shape[0], max_length)
             if padding == 'right':
                 padded[:limit] = vector[:limit]
-            else:  # if padding == 'left
+            else:  # if padding == 'left'
                 padded[max_length - limit:] = vector[:limit]
             return padded
 
         return backend.df_engine.map_objects(ts_vectors, pad)
 
+    @staticmethod
+    def build_matrix_from_column(
+            timeseries,
+            tokenizer_name,
+            length_limit,
+            padding_value,
+            padding,
+            n_steps_ahead,
+            undersample_ratio,
+            missing_value_strategy,
+            backend
+    ):
+        max_length = len(timeseries)
+        if max_length < length_limit:
+            logger.debug(
+                'max length of {0}: {1} < limit: {2}'.format(
+                    tokenizer_name,
+                    max_length,
+                    length_limit
+                )
+            )
+        max_length = length_limit
+        # For the history, we want the most recent point to be n_steps_ahead
+        # back, and then take every undersample_ratio points until reaching
+        # the max length
+        last_index = lambda i: max(0, i - n_steps_ahead)
+        ts_vals = timeseries.values.astype(np.float32)
+        # Iff zeros are appended on the right, put most recent values first
+        rev = 1 if padding == 'right' else -1
+        history = [np.zeros(0, dtype=np.float32) if i < n_steps_ahead else \
+            ts_vals[last_index(i)::-undersample_ratio][:length_limit][::rev] \
+            for i in range(len(timeseries))]
+        ts_vectors = backend.df_engine.df_lib.Series(history)
+
+        # Duplication of logic - not ideal. Also, need error handling.
```

Comment:
You should compute the mean or an other transform parameters in `get_feature_meta ` and then from `build_data` pass the relevant paramenters to this function, so at this point you should already know the mean. Also, this is conflating the padding value with the missing value strategy, maybe there's a better way to handle this and also maybe sending a warning to the user that we are overriding the padding value they specified with the mean is a good idea.

Reply:
Good point. I've added an explicit "padding_value_strategy" option so that the behavior can be explicit, and updated per your suggestion. (Likely the user would usually want this to be the same as the missing value strategy, but right now they are specified independently.)